### PR TITLE
4.7-- code review

### PR DESCRIPTION
**Pull request by savwalts**

### CODE REVIEW

File: db/orderProducts.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        FROM order_products
+        WHERE id=${id};`);
+        return order_products;
+    }catch(error){
+        throw error;
+    }
+};
+
+//needs to update order_product if exists
+const addProductToOrder = async ({orderId, productId, price, quantity}) => {
```
Comment (wallacepreston) [inline db/orderProducts.js:16]: REQUIRED - Please test

File: db/orderProducts.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        INSERT INTO order_products ("orderId", "productId", price, quantity)
+        VALUES ($1, $2, $3, $4)
+        `, [orderId, productId, price, quantity]);
+        return order_products;
+    }catch(error){
+        throw error;
+    }
+};
+
+const updateOrderProduct = async({id, price, quantity}) => {
```
Comment (wallacepreston) [inline db/orderProducts.js]: REQUIRED - Please test

File: db/orderProducts.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        SET price = $1, quantity = $2
+        WHERE id=${id};
+        `, [price, quantity]);
+        return order_products;
+    }catch(error){
+        throw error;
+    }
+};
+
+const destroyOrderProduct = async(id) => {
```
Comment (wallacepreston) [inline db/orderProducts.js:65]: REQUIRED - Please test

File: db/orders.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+const {client} = require('./index');
+const { getUserById } = require('./users');
+
+//----------------------------------------------------
```
Comment (wallacepreston) [inline db/orders.js]: REQUIRED - Remove comments

File: routes/orderProductsRouter.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+const express = require('express');
+const { addProductToOrder, getOrderProductById, updateOrderProduct, destroyOrderProduct } = require('../db/orderProducts');
+const orderProductsRouter = express.Router();
+
+//needs to add order_product to order
+orderProductsRouter.post('/orders/:orderId/products', async(req, res, next) => {
```
Comment (wallacepreston) [inline routes/orderProductsRouter.js]: REQUIRED - Please test

File: routes/orderProductsRouter.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+		console.log('order_product retrieved', order_product);
+		res.send(order_product);
+	}
+	catch(error){
+		next(error);
+	}
+});
+
+//needs to update quantity or price of order_product
+orderProductsRouter.patch('/order_products/:orderProductId', async(req, res, next) => {
```
Comment (wallacepreston) [inline routes/orderProductsRouter.js]: REQUIRED - Please test

File: routes/orderProductsRouter.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    try{
+        const order_product = await updateOrderProduct(id, price, quantity);
+        res.send(order_product);
+    }catch(error){
+        next(error);
+    }
+});
+
+//needs to remove product from order
+orderProductsRouter.delete('/order_products/:orderProductId', async(req, res, next) => {
```
Comment (wallacepreston) [inline routes/orderProductsRouter.js]: REQUIRED - Please test

File: routes/ordersRouter.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	catch(error){
+		next(error);
+	}
+});
+
+ordersRouter.get('/:id', async(req, res, next) => {
+	const { id } = req.params;
+	try{
+		const order = await getOrderById(id);
```
Comment (wallacepreston) [inline routes/ordersRouter.js:45]: REQUIRED - Should be only sending back the order if `order.userId` matches the `req.user.id` (or if you're an admin :)

File: routes/ordersRouter.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const { getAllOrders, getOrderById, getOrdersByUser } = require('../db/orders');
+const ordersRouter = express.Router();
+
+ordersRouter.get('/', async(req, res, next) => {
```
Comment (wallacepreston) [inline routes/ordersRouter.js]: REQUIRED - Should be only sending back if `req.user.isAdmin` (or something)

File: routes/utils.js [hunk 10]
```diff
@@ -1,10 +1,31 @@
+const jwt = require('jsonwebtoken');
+const { JWT_SECRET } = process.env;
+
+
 const requireUser = (req, res, next) => {
```
Comment (wallacepreston) [inline routes/utils.js:1]: REQUIRED 
- Some of this functionality  (the bulk of it, the token verify etc) should be added to a middleware `req.use` in the `routes/index.js` file before all API routers get attached.
- Then this function should just check if there is a `req.user` defined.

File: routes/usersRouter.js [hunk 11]
```diff
@@ -9,41 +9,80 @@ const {
...
   getAllUsers,
+  getUserById,
 } = require('../db/users');
 
 const {
 	createUser,
 } = require('../db/index');
 
 
+usersRouter.get('/', async (req, res, next) => {
```
Comment (wallacepreston) [inline routes/usersRouter.js]: REQUIRED - Should be only accessible by admin users.

File: routes/utils.js [hunk 12]
```diff
@@ -1,10 +1,31 @@
+const jwt = require('jsonwebtoken');
+const { JWT_SECRET } = process.env;
```
Comment (wallacepreston) [inline routes/utils.js]: REQUIRED - Should have a default value

**Review by wallacepreston — CHANGES_REQUESTED** [review]
Please fix the "REQUIRED"s and we'll review again!

File: db/orderProducts.js (hunk 1 above)
Comment (MatthewRoda) [reply db/orderProducts.js:16]: Tested and functioning!

File: db/orderProducts.js (hunk 2 above)
Comment (MatthewRoda) [reply db/orderProducts.js]: Tested, edited, now functioning with code to be pushed.

File: db/orderProducts.js (hunk 3 above)
Comment (MatthewRoda) [reply db/orderProducts.js:65]: Tested, functioning as is.

File: routes/orderProductsRouter.js (hunk 5 above)
Comment (MatthewRoda) [reply routes/orderProductsRouter.js]: tested, edited, changes pushed to dev!

File: routes/utils.js (hunk 10 above)
Comment (MatthewRoda) [reply routes/utils.js:1]: Edited to meet this criteria

File: routes/utils.js (hunk 12 above)
Comment (MatthewRoda) [reply routes/utils.js]: This is no longer present in this file.

File: db/orders.js (hunk 4 above)
Comment (MatthewRoda) [reply db/orders.js]: removed

File: routes/ordersRouter.js (hunk 8 above)
Comment (MatthewRoda) [reply routes/ordersRouter.js:45]: edited, tested, functioning as requested.

File: routes/ordersRouter.js (hunk 9 above)
Comment (MatthewRoda) [reply routes/ordersRouter.js]: requireAdmin applied.

File: routes/usersRouter.js (hunk 11 above)
Comment (MatthewRoda) [reply routes/usersRouter.js]: requireAdmin applied.

**Comment by MatthewRoda** [comment]
All required changes were made and tested!

**Review by wallacepreston — APPROVED** [review]
Release 0.5 Delta

**Pull request merged**